readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HdfsOffsetComputer.java

```diff
@@ -8,82 +8,78 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
 import java.io.IOException;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 import java.util.function.Function;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 /**
  * Persist and read offset based on HDFS file names, named _date_/_partition_._offset_
  */
 public class HdfsOffsetComputer implements OffsetComputer {
     private static final Logger LOGGER = LoggerFactory.getLogger(HdfsOffsetComputer.class);
-    private final Function<Integer, Pattern> offsetFilePatternGenerator;
+    private final Pattern offsetFilePatternGenerator;
     private final FileSystem fs;
     private final Path basePath;
+    private final String kafkaCluster;
+
+    public HdfsOffsetComputer(FileSystem fs, Path basePath) {
+        this(fs, basePath, null);
+    }
 
     /**
      * @param fs                            Filesystem for said files
      * @param basePath                      Root directory to look in for offsets filenames
+     * @param kafkaCluster                  Name of corresponding kafka cluster
      */
-    public HdfsOffsetComputer(FileSystem fs, Path basePath) {
+    public HdfsOffsetComputer(FileSystem fs, Path basePath, @Nullable String kafkaCluster) {
         this.fs = fs;
         this.basePath = basePath;
-        this.offsetFilePatternGenerator = partitionId -> Pattern.compile(String.format("^%d.(\\d+)$", partitionId));
+        this.kafkaCluster = kafkaCluster;
+        if (kafkaCluster == null) {
```

Comment:
Can we avoid using null and create a third constructor that injects offsetFilePatternGenerator ?

Reply:
I would avoid putting pattern outside of this class because previous file pattern is tied to file rename pattern.
kafkaCluster may be null because it comes from kafkaConfiguration which is parsed by jackson-databind, hence may be null if not set in the config.

Reply:
ok